data/registry_model/interface.py

```diff
@@ -310,6 +310,13 @@ def get_repo_blob_by_digest(self, repository_ref, blob_digest, include_placement
         the return value of this function may change.
         """
 
+    @abstractmethod
+    def blob_exists(self, model_cache, namespace_name, repo_name, blob_digest):
+        """
+        Returns the blob if it exists in the given namespace, or None if either
+        the blob or repository don't exist.
+        """
```

Comment:
In the proxy cache code we need to distinguish between checking for a blob existence vs actually getting the blob. This method makes that distinction possible.
We need this in proxy code because we don't want to download a blob from the upstream registry when a client is merely checking for its existence (HEAD), but only when the client is actually downloading the blob (GET).